task/mgmt-gateway/src/main.rs

```diff
@@ -120,17 +122,20 @@ fn main() {
 
 struct UsartHandler {
     usart: Usart,
-    to_tx: ArrayVec<[u8; SerialConsole::MAX_DATA_PER_PACKET]>,
-    from_rx: ArrayVec<[u8; SerialConsole::MAX_DATA_PER_PACKET]>,
+    to_tx: &'static mut ArrayVec<SerializedMessageBuf>,
+    from_rx: &'static mut ArrayVec<SerializedMessageBuf>,
     from_rx_flush_deadline: Option<u64>,
 }
 
 impl UsartHandler {
-    fn new(usart: Usart) -> Self {
+    fn new(
+        usart: Usart,
+        [to_tx, from_rx]: &'static mut [ArrayVec<SerializedMessageBuf>; 2],
```

Comment:
I'm not a huge fan of destructuring in function arguments, but that's just, like, _my opinion_, man.

Reply:
Hah, me neither. IIRC this is a relic of trying to get `cargo fmt` to do something reasonable with a long line before I introduced the `SerializedMessageBuf` typedef; it was doing something heinous like

```rust
buffers: &'static mut [ArrayVec<[u8; SOME_LONG_CONSTANT]>;
                       2],
```

With the typedef I don't think that's a concern any more, so I'll switch this back.